client/components/data/theme-details/index.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import omit from 'lodash/object/omit';
+
+/**
+ * Internal dependencies
+ */
+import { fetchThemeDetails } from 'state/themes/actions';
+import { getThemeDetails } from 'state/themes/theme-details/selectors';
+
+/**
+ * Fetches details for a theme specified by its ID
+ * and passes it to the supplied child component.
+ */
+const ThemeDetailsData = React.createClass( {
+
+	propTypes: {
+		children: React.PropTypes.element.isRequired,
+		id: React.PropTypes.string.isRequired,
+		// Connected props
+		name: React.PropTypes.string,
+		author: React.PropTypes.string,
+		fetchThemeDetails: React.PropTypes.func.isRequired
+	},
+
+	componentDidMount() {
+		this.refresh( this.props );
+	},
+
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.id && nextProps.id !== this.props.id ) {
+			this.refresh( nextProps );
+		}
+	},
+
+	refresh( props ) {
```

Comment:
Could maybe take `id` rather than `props` as a param to make the body slightly simpler.